### PR TITLE
Build test container on PR

### PR DESCRIPTION
**Pull request by jbygdell**
This will simplify testing of PRs

### CODE REVIEW

File: .github/workflows/build_pr_container.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+      - name: Build container
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: ./Dockerfile
+          push: true
+          tags: |
+            ghcr.io/${{ github.repository }}:PR${{ github.event.number }}
+            ghcr.io/${{ github.repository }}:sha-${{ github.sha }}
```
Comment (nanjiangshu) [inline .github/workflows/build_pr_container.yaml:34]: Why do we need two tags for the same image? Is that for when `{{ github.event.number }}` returns an empty string?

**Review by nanjiangshu — COMMENTED** [review]
This workflow looks good and works well when I run it locally with `act`.
However, I don't know exactly how to use the PR container for testing. Do you have any good examples and maybe good to document it somewhere.

File: .github/workflows/build_pr_container.yaml (hunk 1 above)
Comment (jbygdell) [reply .github/workflows/build_pr_container.yaml:34]: A forced push will change the `sha` but not the PR number

**Comment by jbygdell** [comment]
> This workflow looks good and works well when I run it locally with `act`. However, I don't know exactly how to use the PR container for testing. Do you have any good examples and maybe good to document it somewhere.

This will allow us to test the actual PR container in any test environment be it local or a test cluster.

**Review by aaperis — APPROVED** [review]

File: .github/workflows/build_pr_container.yaml (hunk 1 above)
Comment (nanjiangshu) [reply .github/workflows/build_pr_container.yaml:34]: Good point.

**Review by nanjiangshu — APPROVED** [review]

**Review by dbampalikis — APPROVED** [review]

**Pull request merged**